pep8.py

```diff
@@ -1846,7 +1846,15 @@ def check_files(self, paths=None):
         report.start()
         try:
             for path in paths:
-                if os.path.isdir(path):
+                if path == "-":
+                    # STDIN
+                    path = self.options.stdin_name
+                    # if no ersatz filename for STDIN, assume not excluded
+                    excluded = path and self.excluded(path)
+                    if not excluded:
+                        runner(path or "stdin",
+                               lines=stdin_get_value().splitlines(True))
```

Comment:
I just followed the lead of the `Checker` class, which contains this code in its `__init__()`:

```
        elif filename == '-':
            self.filename = 'stdin'
            self.lines = stdin_get_value().splitlines(True)
```

The reason it's being special cased here is that it's the first place in the code where we're examining the `path` variable, and it felt wrong to me to do the equivalent of `os.path.isdir('-')` as the first check.

the `isdir()` check usually fails and the `-` path would also fail the `excluded()` check and then be passed into the `Checker` and trigger the code above.

This assumes there isn't an actual `-` directory at the current working directory, so this PR actually fixes a hidden bug, I think (haven't tested this case).

On the other hand, if we pass in the lines explicitly into the checker (using the same code that `Checker` uses to split `stdin`), AND pass `None` as the filename, we hit this other branch of the `Checker` code:

```
        if filename is None:
            self.filename = 'stdin'
            self.lines = lines or []
```

Which means that `Checker` won't try to read the lines by itself, AND will assume the name is `stdin`.

Perhaps we should refactor this one line snippet somewhere, explain it and make it official, but I also don't know why `Checker` keeps the line endings :-)

Also, perhaps `Checker` should not be in the business of reading files and splitting lines, but this is not a refactor I have time to do right now :-)